### PR TITLE
Nlj jvb stats

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: src/main/java/org/jitsi/videobridge/rest/HandlerImpl.java [hunk 1]
```diff
@@ -362,6 +366,23 @@ else if (DOMINANT_SPEAKER_IDENTIFICATION.equals(target))
         }
     }
 
+    /**
+     * Gets the handler for statistics requests.
+     * @return
+     */
+    private StatisticsRequestHandler getStatisticsRequestHandler()
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/rest/HandlerImpl.java]: do we have to worry about synchronization here?

File: src/main/java/org/jitsi/videobridge/rest/StatisticsRequestHandler.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        if ("GET".equals(request.getMethod()))
+        {
+            JSONObject statsJsonObject = new JSONObject();
+            if (target.equals(STATISTICS))
+            {
+                statsJsonObject = doGetStatisticsJSON();
+            }
+            else if (target.equals(STATISTICS + "/" + NODE))
+            {
+                // TODO
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/rest/StatisticsRequestHandler.java]: i thought you had a lot of these working already?

File: src/main/java/org/jitsi/videobridge/rest/StatisticsRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+                return;
+            }
+
+            Boolean all = getBoolean(requestJSONObject.get(ALL));
+            Boolean node = getBoolean(requestJSONObject.get(NODE));
+            Boolean queue = getBoolean(requestJSONObject.get(QUEUE));
+            Boolean pool = getBoolean(requestJSONObject.get(POOL));
+            Boolean transit = getBoolean(requestJSONObject.get(TRANSIT));
+
+            if (node != null || (node = all) != null)
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/rest/StatisticsRequestHandler.java]: what does the url look like to enable/disable stats?

File: src/main/java/org/jitsi/videobridge/util/PartitionedByteBufferPool.java [hunk 4]
```diff
@@ -16,171 +16,362 @@
...
+     * a given initial size for each partition.
+     */
+    PartitionedByteBufferPool()
     {
         for (int i = 0; i < NUM_PARTITIONS; ++i)
         {
-            partitions[i] = new Partition(i, initialSize);
+            partitions[i] = new Partition(i, INITIAL_SIZE);
         }
+        logger.info("Initializes a new " + getClass().getSimpleName()
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/util/PartitionedByteBufferPool.java]: nit: `Initializes` -> `Initialized`

File: src/main/java/org/jitsi/videobridge/util/PartitionedByteBufferPool.java [hunk 5]
```diff
@@ -16,171 +16,362 @@
...
-                    }
-                } else {
-                    if (ByteBufferPool.ENABLE_BOOKKEEPING)
-                    {
-                        numNoAllocationNeeded.incrementAndGet();
-                    }
-                    break;
+            if (enableStatistics)
+            {
+                numRequests.incrementAndGet();
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/util/PartitionedByteBufferPool.java]: so here we only increment numRequests if stats is enabled, but at the top level i think we always do.  is that intentional?

File: src/main/java/org/jitsi/videobridge/util/PartitionedByteBufferPool.java [hunk 6]
```diff
@@ -16,171 +16,362 @@
...
+                // the buffer to be too small in practice, we'll throw it away.
+                // This makes sure that if someone returns a small buffer (in
+                // the practical sense) it will not get stuck in the pool.
+                if (buf.length >= DEFAULT_BUFFER_SIZE)
+                {
+                    pool.offer(buf);
+                }
+                else
+                {
+                    numSmallBuffersDiscarded.incrementAndGet();
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/util/PartitionedByteBufferPool.java]: should we consider `SMALL_BUFFERS_ACCEPTED` here?

**Comment by webintoit** [comment]
> _No description provided._
What is NLJ means?

**Comment by bbaldino** [comment]
@webintoit `nlj` refers to 'new libjitsi', which was an early nickname for what is now the `jitsi-media-transform` repo

File: src/main/java/org/jitsi/videobridge/rest/StatisticsRequestHandler.java (hunk 3 above)
Comment (bgrozev) [reply src/main/java/org/jitsi/videobridge/rest/StatisticsRequestHandler.java]: Example: 
POST
{ "all" : false, "pool": true}
To disable everything and enable the pool stats.

**Pull request merged**